gender_novels/deployment/static/markdowns/metadata_acquisition.md

```diff
@@ -0,0 +1,119 @@
+
+	if (not is_valid_novel_gutenberg(gutenberg_id)):
+		print("Not a novel")
+		continue
+	novel_metadata = get_gutenberg_metadata_for_single_novel(gutenberg_id)
+	write_metadata(novel_metadata)
+	...
+	with open(Path(BASE_PATH, 'corpora', 'gutenberg', 'texts', f'{gutenberg_id}.txt'),
+		mode='w', encoding='utf-8') as outfile:
+		outfile.write(text_clean)
+		number_books += 1
+
+In the process of generating the corpus, we found that we had more than enough novels.  So greater emphasis was put on ensuring that included books were valid novels, even if it led to the exclusion of some legitimate novels.  We checked novels for 
+
+ - Language and copyright status  using the gutenberg module
+ - Certain keywords in the Gutenberg-defined subjects that indicated nonfiction or periodicals, like `correspondence`.  All novels had to contain the word `fiction` in their subject categories.  
+ - Publication dates between 1770 and 1922
+ - Certain phrases in the titles like `Index of the Project Gutenberg Works of`...
+ - Indications that the text was a translation, like the presence of the string `Translator: ` in the text
+ - The length of the text: all novels had to be between 140000 and [9609000](http://www.guinnessworldrecords.com/world-records/longest-novel) characters.  
+
+For the methods that actually acquired the metadata, we used a process like this:
+
+	date = get_publication_date_from_copyright_certain(novel_text)
+	if date:
+		return date
+	else:
+		date = get_publication_date_wikidata(author, title)
+	if not date:
+		date = get_publication_date_from_copyright_uncertain(novel_text)
+	return date
+
+Essentially, we would first try to get metadata from the most reliable source (in this case, the book itself), and if that failed we would try every method from most reliable to least reliable in turn, before giving up.  (For example, if the Library of Congress metadata acquisition was implemented, it would take first precedence).  
+
+In several hours with a good computer it is possible to go through all the books from Project Gutenberg and generate a corpus with metadata.  
+
+### A possible improvement
+
+The `is_valid_novel` function needs to look up the publication date, so it calls `get_publication_date`.  Let's say `get_publication_date` manages to get the date from Wikidata, but it still loads the novel's text by `get_novel_text_gutenberg` for the `get_publication_date_from_copyright` function.  Then later, when evaluating the text, `is_valid_novel`calls `get_novel_text_gutenberg` again to check the length.  Then, once all that is done, `get_publication_date_from_copyright` gets called again to fill in the metadata field for the publication date, which once again calls `get_novel_text_gutenberg` and also `get_publication_date_wikidata`...  Each loop we call many functions more than once, and this happens roughly 50,000 times.  
```

Comment:
When writing functions inline, I would include the parentheses, e.g. `is_valid_novel()`, `get_publication_date()`, etc. It signals to the reader that those are functions without too much extra explanation.